corehq/apps/domain/models.py

```diff
@@ -622,6 +623,47 @@ def get_or_create_with_name(cls, name, is_active=False,
             new_domain.save(**get_safe_write_kwargs())
             return new_domain
 
+    @classmethod
+    def generate_name(cls, hr_name, max_length=25):
+        '''
+        Generate a URL-friendly name based on a given human-readable name.
+        Normalizes given name, then looks for conflicting domains, addressing
+        conflicts by adding "-1", "-2", etc. May return None if it fails to
+        generate a new, unique name.
+        '''
+
+        original_name = hr_name.strip().lower()
+        original_name = re.sub(r'[^0-9a-z]+', '-', original_name)
+
+        def has_conflict(name):
+            return Domain.get_by_name(name) or Domain.get_by_name(name.replace('-', '.'))
+
+        name = original_name
+        shorten = -1
+        while has_conflict(name) and shorten + 2 < max_length:
+            shorten = shorten + 1
+            name_prefix = original_name
+            if shorten > 0:
+                name_prefix = name_prefix[:-shorten]
+            hits = Domain.get_names_by_prefix(name_prefix + "-")
+            if re.search(r'\.', name_prefix):
+                hits += Domain.get_names_by_prefix(name_prefix.replace('-', '.') + "-")
```

Comment:
I originally took that because that's how the orgs app checks for name conflicts: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/orgs/forms.py#L24

...but, re-reading, that's dealing with hr_name, not name, so I can drop this code.